platform/ui/src/components/Dropdown/Dropdown.tsx

```diff
@@ -5,16 +5,48 @@ import classnames from 'classnames';
 import Icon from '../Icon';
 import Typography from '../Typography';
 
-const Dropdown = ({ id, children, showDropdownIcon, list, titleClassName }) => {
+const borderStyle = 'border-b last:border-b-0 border-secondary-main';
+
+const Dropdown = ({
+  id,
+  children,
+  showDropdownIcon,
+  list,
+  itemsClassName,
+  titleClassName,
+  showBorders,
+  alignment,
+  // By default the max characters per line is the longest title
+  // if you wish to override this, you can pass in a number
+  maxCharactersPerLine,
+}) => {
   const [open, setOpen] = useState(false);
   const element = useRef(null);
 
+  // choose the max characters per line based on the longest title
+  const longestTitle = list.reduce((acc, item) => {
+    if (item.title.length > acc) {
+      return item.title.length;
+    }
+    return acc;
+  }, 0);
+
+  maxCharactersPerLine = maxCharactersPerLine ?? longestTitle;
+
   const DropdownItem = useCallback(({ id, title, icon, onClick }) => {
+    // Split the title into lines of length maxCharactersPerLine
+    const lines = [];
+    for (let i = 0; i < title.length; i += maxCharactersPerLine) {
```

Comment:
I mean what choice do we have, it is either a very wide dropdown as a result of only one of the items being longs or break like this. In practice I have not seen this happening

Reply:
Well because it is an edge case we should leave it, but I suspect there would be something we could do with css https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_text/Wrapping_breaking_text